src/test/java/io/airlift/compress/AbstractTestCompression.java

```diff
@@ -174,6 +174,13 @@ public void testCompress(DataSet testCase)
         byte[] originalUncompressed = testCase.getUncompressed();
         byte[] compressed = new byte[compressor.maxCompressedLength(originalUncompressed.length)];
 
+        // attempt to compress slightly similar data to ensure the compressor doesn't keep state
```

Comment:
It's offset by one position (and the first byte is removed). If the compressor keeps the hash table around, it's likely that it will see a bunch of matches, but all with the wrong offsets.